pkg/minikube/tunnel/tunnel.go

```diff
@@ -0,0 +1,160 @@
+/*
+Copyright 2018 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tunnel
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/machine/libmachine"
+	"github.com/golang/glog"
+	"github.com/pkg/errors"
+	"k8s.io/client-go/kubernetes/typed/core/v1"
+	"k8s.io/minikube/pkg/minikube/config"
+)
+
+type tunnel interface {
+	cleanup() *Status
+	updateTunnelStatus() *Status
+}
+
+func errorTunnelAlreadyExists(id *ID) error {
+	return fmt.Errorf("there is already a running tunnel for this machine: %s", id)
+}
+
+func newTunnel(machineName string,
```

Comment:
Keep function signatures as a single line when possible. When it isn't possible, there is your hint that you should take a config struct.